src/Esprima/Utils/AstJson.cs

```diff
@@ -0,0 +1,830 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions;
+using Esprima.Ast;
+
+namespace Esprima.Utils
+{
+    public enum LocationMembersPlacement
+    {
+        End,
+        Start,
+    }
+
+    public static class AstJson
+    {
+        public sealed class Options
+        {
+            public static readonly Options Default = new Options();
+
+            public bool IncludeLineColumn { get; private set; }
+            public bool IncludeRange { get; private set; }
+            public LocationMembersPlacement LocationMembersPlacement { get; private set; }
+
+            public Options() {}
+
+            Options(Options options)
+            {
+                IncludeLineColumn = options.IncludeLineColumn;
+                IncludeRange = options.IncludeRange;
+                LocationMembersPlacement = options.LocationMembersPlacement;
+            }
+
+            public Options WithIncludeLineColumn(bool value) =>
+                value == IncludeLineColumn ? this : new Options(this) { IncludeLineColumn = value };
+
+            public Options WithIncludeRange(bool value) =>
+                value == IncludeRange ? this : new Options(this) { IncludeRange = value };
+
+            public Options WithLocationMembersPlacement(LocationMembersPlacement value) =>
+                value == LocationMembersPlacement ? this : new Options(this) { LocationMembersPlacement = value };
+
+            public Options WithLineColumn() => WithIncludeLineColumn(true);
```

Comment:
> Jint has options with `WithX(bool value = true)`

Do you have examples? BTW, a default of non-default is usually inviting trouble.

`WithLineColumn` is a mere helper and I can remove it if it's confusing.

@sebastienros Thoughts?

Reply:
Here's couple examples from [Jint's Options](https://github.com/sebastienros/jint/blob/b5c4b570cea2ac34f6be3ce67ccb737b9fa787ba/Jint/Options.cs#L32-L45)

Reply:
That's not the same design. Jint's `Options` is multable and has helpers designed to support a fluent syntax at the call site. The `Options` here uses an immutable design (like [records](https://github.com/dotnet/csharplang/blob/master/proposals/records.md)) where `WithX` members are designed to return modified copies (or the same instance if no modification is necessary).

Reply:
I would just have `IncludeXXX(bool)` because it's a verb, and `WithStuff(bool)` when `Stuff` is a noun. Also I agree that `WithLineColumn` is missleading.

Reply:
Addressed with d46e9e88905542f53d6d73dc90b56466bd63388d and 32f1703b5b0391c43a0ca7af99d018b716de58e6.